pkg/testutils/testutils.go

```diff
@@ -66,6 +71,17 @@ func CreateFakeRequestJSON(uid string,
 			},
 		},
 	}
+	// Allow for additional optional parameters
```

Comment:
What are these for?

Reply:
I've added an optional parameter (`opts`) to allow for a test to pass in Name and/or Namespace to the request. For NetNamespaces, we are checking if it's a valid Name.

Reply:
I gather that from the code, but what issue is being addressed by the change? How are they used by the API server with an incoming payload, and why is it different than the payload in the `object`?

Reply:
The problem is that the current test utils do not allow for tests to set the Name and/or Namespace in the request. As defined in the [AdmissionRequest](https://github.com/kubernetes/api/blob/master/admission/v1beta1/types.go#L92):
```
        // Name is the name of the object as presented in the request.  On a CREATE operation, the client may omit name and
	// rely on the server to generate the name.  If that is the case, this field will contain an empty string.
	// +optional
	Name string `json:"name,omitempty" protobuf:"bytes,5,opt,name=name"`
```
Adding optional parameters to this function allows tests to both set the Name and Request.Name.
The webhook auth code is checking against the Request.Name, not the object or oldObject name.

Reply:
I've updated the webhook and test code to check the `Request.Object.Name` as opposed to checking the `Request.Name` to limit the amount of churn in the testutils code.

Reply:
Can you help me understand why we can't rely on the contents in the incoming`AdmissionRequest`'s `Object` (or `OldObject`, verb depending)? I do not want to complicate things more unless it's a requirement.

It would help if you could post the incoming request into this webhook for each verb (perhaps obfuscating the `userInfo` section).

Reply:
As I've said above, I've changed the code to rely on the AdmissionRequest Object, not the Name as to not introduce complexity to the code. Please see the updated code. Thanks.